lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -460,9 +431,19 @@ private boolean achievedWithGpg45Profile(
             Vot requestedVot,
             List<SignedJWT> credentials,
             List<VcStoreItem> vcStoreItems,
-            AuditEventUser auditEventUser)
-            throws UnknownEvidenceTypeException, ParseException, SqsException {
-
+            AuditEventUser auditEventUser,
+            boolean isF2FComplete)
+            throws UnknownEvidenceTypeException, ParseException, SqsException,
+                    HttpResponseExceptionWithErrorBody, CredentialParseException {
+        // Check for credential correlation failure
+        if (!userIdentityService.areVCsCorrelated(vcStoreItems)) {
+            if (isF2FComplete) {
+                sendAuditEvent(AuditEventTypes.IPV_F2F_CORRELATION_FAIL, auditEventUser);
+            } else {
+                sendAuditEvent(AuditEventTypes.IPV_IDENTITY_REUSE_RESET, auditEventUser);
```

Comment:
I think this is still not _quite_ right - you might still match an operational profile after this (in which case we don't want these events).

I believe we should record:
- IPV_F2F_CORRELATION_FAIL if f2f is complete, your VCs don't correlate, and you don't match _any_ profile
- IPV_F2F_PROFILE_NOT_MET_FAIL if f2f is complete, your VCs do correlate, but you don't match any profile
- IPV_IDENTITY_REUSE_RESET if it's not f2f, and you don't match any profile

I can think of two approaches:
- Pull the correlation check back out to the top level, but store in a variable `areGpg45VcsCorrelated`
- Pass that variable into `checkProfileMatch` and skip GPG45 profiles if correlation fails
- use the combination of `isF2FComplete` and `areGpg45VcsCorrelated` to decide which audit and journey events to log afterwards

or

- Keep the correlation check here, and re-check correlation if you fail to match a profile with f2f complete - has some duplication, but might be easier to implement/understand

Reply:
With above change I tried to keep all the audit events as it is.
Here we can only log audit event IPV_F2F_CORRELATION_FAIL if f2f is complete.

With above statement - use the combination of isF2FComplete and areGpg45VcsCorrelated to decide which audit and journey events to log afterwards.
R u suggesting doing this in achievedWithGpg45Profile().